packages/app/src/hooks/useExplainQuery.tsx

```diff
@@ -0,0 +1,36 @@
+import { useQuery } from '@tanstack/react-query';
+
+import { sendQuery } from '@/clickhouse';
+import {
+  ChartConfigWithDateRange,
+  renderChartConfig,
+} from '@/renderChartConfig';
+
+export function useExplainQuery(
+  config: ChartConfigWithDateRange,
+  options?: { enabled?: boolean },
+) {
+  const {
+    data: explainData,
+    isLoading,
+    error,
+  } = useQuery({
+    queryKey: ['explain', config],
+    queryFn: async ({ signal }) => {
+      const query = await renderChartConfig(config);
+      const response = await sendQuery<'JSONEachRow'>({
+        query: `EXPLAIN ESTIMATE ${query.sql}`,
+        query_params: query.params,
+        format: 'JSONEachRow',
+        abort_signal: signal,
+        connectionId: config.connection,
+      });
+      return response.json();
+    },
+    enabled: options?.enabled,
+    retry: false,
+    staleTime: 1000 * 60,
+  });
+
+  return { explainResults: explainData, isLoading, error };
```

Comment:
more of a style note: my pref usually is to return the react hook results as-is and let the consumer rename/grab the data as it sees fit.

Reply:
Ah thanks for catching that.